### ISSUE TITLE
Why doesn't kfold_predict throw an error with new population factor levels?

### ISSUE THREAD

**Issue by gcolitti** [paul-buerkner/brms#602]
I'm likely missing something obvious, but I can't figure out why `kfold_predict` returns predictions for new population factor levels. Shouldn't this throw an error in the same way that `predict.brmsfit` does?

**Comment by paul-buerkner** [comment]
Do you have a reproducible example for me?

Am Di., 12. Feb. 2019, 19:16 hat gcolitti <notifications@github.com>
geschrieben:

**Comment by gcolitti** [comment]
`require(brms)

d <- as.data.frame(cbind(y = sample(1:100, 10), x = rep(1:10, 1)))
d$x <- as.factor(d$x)

# No error using kfold_predict even though there are only 10 obs and 10 levels
# for factor x.
fit <- brm(data = d, formula = y ~ 1 + x)
kfold_obj <- kfold(fit, K = 2, save_fits = TRUE)
kfold_predictions <- kfold_predict(kfold_obj)

# Throws an error when using predict.brms.
fit2 <- brm(data = d[-1, ], formula = y ~ 1 + x)
predict(fit2, newdata = d[1, ])`

**Comment by paul-buerkner** [comment]
You don't pass any `newdata` to `kfold`. Why should it complain?

**Comment by gcolitti** [comment]
Maybe I just don't understand what `kfold` and `kfold_predict` are doing. If I use `kfold_obj <- kfold(fit, K = 2, save_fits = TRUE, newdata = d)`  I get the complaint about new factor levels. But doesn't `kfold` use the original data used to fit the `brm` model by default? So why do I need to use `newdata = d`? Also, if `kfold` just partitions the data into K random subsets and fits the model K times, why does `kfold` return the error and not `kfold_predict`?

**Comment by paul-buerkner** [comment]
You are right, `kfold` should complain as well even if the original data are used, since in the test set, there are levels of `x` not in the training set. I will fix that.

`kfold_predict` is just a thin function basically taking the results of `kfold` and make them look differently so that people can work with it. The actually prediction already happend in `kfold`.

**Comment by paul-buerkner** [comment]
I just fixed the issue in the github version. Thanks for bringing this to my attention!

**Comment by gcolitti** [comment]
I love the package. Thank you!
